[BUG] nvcomp_adapter fails to compile if cudf logging is disabled

**Describe the bug**
After #17431 libcudf fails to compile when cudf logging is disabled with an error like the following:
```
[INFO]      [exec] /home/jlowe/src/spark-rapids-jni/thirdparty/cudf/cpp/src/io/comp/nvcomp_adapter.cpp:73:13: error: 'std::string cudf::io::nvcomp::{anonymous}::compression_type_name(cudf::io::nvcomp::compression_type)' defined but not used [-Werror=unused-function]
[INFO]      [exec]    73 | std::string compression_type_name(compression_type compression)
[INFO]      [exec]       |             ^~~~~~~~~~~~~~~~~~~~~
[INFO]      [exec] cc1plus: all warnings being treated as errors
```

**Steps/Code to reproduce bug**
Compile libcudf with -DLIBCUDF_LOGGING_LEVEL=OFF

**Expected behavior**
libcudf can compile with logging disabled

**Environment overview (please complete the following information)**
 - Environment location: Docker, see Dockerfile and build-in-docker in https://github.com/NVIDIA/spark-rapids-jni
 - Method of cuDF install: from source

--- a/cpp/src/io/comp/nvcomp_adapter.cpp
+++ b/cpp/src/io/comp/nvcomp_adapter.cpp
@@ -70,7 +70,7 @@ auto batched_decompress_async(compression_type compression, Args&&... args)
   }
 }
 
-std::string compression_type_name(compression_type compression)
+[[maybe_unused]] std::string compression_type_name(compression_type compression)
 {
   switch (compression) {
     case compression_type::SNAPPY: return "Snappy";

